addons/im_livechat/static/src/embed/common/livechat_service.js

```diff
@@ -181,20 +182,24 @@ export class LivechatService {
     _saveLivechatState(threadData, { persisted = false } = {}) {
         const { guest_token } = threadData;
         if (guest_token) {
-            localStorage.setItem(GUEST_TOKEN_STORAGE_KEY, threadData.guest_token);
+            expirableStorage.setItem(GUEST_TOKEN_STORAGE_KEY, threadData.guest_token);
             delete threadData.guest_token;
         }
-        cookie.set(
-            SAVED_STATE_COOKIE,
+        expirableStorage.setItem(
+            SAVED_STATE_STORAGE_KEY,
             JSON.stringify({
                 threadData: persisted ? { id: threadData.id, model: threadData.model } : threadData,
                 persisted,
                 livechatUserId: this.savedState?.livechatUserId ?? session.user_id,
             }),
-            60 * 60 * 24
+            60 * 60 * 24 // kept for 1 day.
         );
         if (threadData.operator) {
-            cookie.set(OPERATOR_COOKIE, threadData.operator.id, 7 * 24 * 60 * 60);
+            expirableStorage.setItem(
+                OPERATOR_STORAGE_KEY,
+                threadData.operator.id,
+                7 * 24 * 60 * 60 // kept for 7 days.
+            );
```

Comment:
Note that the comments `kept for X` could have been prevented if the function signature allowed options in a better shape, e.g. `{ days: 7 }`. This could be improved in `master` eventually.

Reply:
Agreed, I did this to keep previous values intact but your suggestion is much better :)